song-server/src/main/java/org/icgc/dcc/song/server/service/AnalysisService.java

```diff
@@ -50,21 +49,43 @@
   @Autowired
   private final IdService idService;
   @Autowired
-  private final CompositeEntityService sampleService;
+  private final CompositeEntityService compositeEntityService;
+  @Autowired
+  private final SampleService sampleService;
+
   @Autowired
   private final FileService fileService;
   @Autowired
   private final ExistenceService existence;
 
   @SneakyThrows
+  public String update(String studyId, Analysis a) {
+    val id=a.getAnalysisId();
+    System.err.printf("WARNING: NOT UPDATING analysis id '%s': code not finished", id);
+    return id;
+  }
+
+  public String save(@NonNull String studyId, @NonNull Analysis analysis) {
+    val submitters = new ArrayList<String>();
+    analysis.getSample().forEach(s->submitters.add(s.getSampleSubmitterId()));
+    String id = findByBusinessKey(studyId, analysis.getAnalysisType(), submitters);
+
+    if (id == null) {
```

Comment:
instead of using null for verifying id was undefined, use the Optional class. This makes it clear that id is _intended_ to defined or undefined.
```
Optional<String> optionalId = findByBusinessKey(.....) //need to update findByBusinessKey to return Optional<String>
String id;
if (optionalId.isPresent()) {
    update(studyId, analysis)
}  else {
    return create(studyId, analysis)
}
return optionalId.get();
```